src/player.h

```diff
@@ -73,6 +73,18 @@ struct special_attack {
     };
 };
 
+class player_morale;
+class player_morale_ptr : public std::unique_ptr<player_morale> {
```

Comment:
Agreed, but I couldn't reuse the existing template because it lacks of necessary constructors/assignment operators. When I was trying to turn my class into template, it was a struggle with the compiler and it won. It refuses to manage memory allocation/deallocation on an unfinished type and I decided to leave the class for now. I'll come back to this later and try to develop a generic approach.